src/engine/controls/macrocontrol.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include <rigtorp/SPSCQueue.h>
+
+#include <QString>
+
+#include "control/controlpushbutton.h"
+#include "engine/controls/enginecontrol.h"
+#include "track/macro.h"
+
+class MacroControl : public EngineControl {
+    Q_OBJECT
+  public:
+    MacroControl(const QString& group, UserSettingsPointer pConfig, int slot);
+
+    void trackLoaded(TrackPointer pNewTrack) override;
+    void process(const double dRate,
+            mixxx::audio::FramePos currentPosition,
+            const int iBufferSize) override;
+    void notifySeek(mixxx::audio::FramePos position) override;
+
+    bool isRecording() const;
+
+    enum class Status : int {
+        NoTrack = -1,
+        Empty = 0,
+        Armed = 1,
+        Recording = 2,
+        Recorded = 3,
+        Playing = 4,
+    };
```

Comment:
```suggestion
    enum class Status : int {
        /// No track is loaded.
        NoTrack = -1,
        /// There is no recoded macro present.
        Empty = 0,
        /// Recording is armed. The next supported action (e.g., a hotcue jump will start the recording).
        Armed = 1,
        /// Recording has been started and supported actions will be part of the resulting macro.
        Recording = 2,
        /// The is a recorded macro present, but it's not enabled for playback.
        Recorded = 3,
        /// The macro is currently enabled for playback.
        Playing = 4,
    };
```

We could rename `Armed` to `RecordingArmed` to make this clearer?